core-bundle/src/Framework/ContaoFramework.php

```diff
@@ -148,6 +149,19 @@ public static function getNonce(): string
         return self::$nonce;
     }
 
+    public function setLocale(string $locale): void
```

Comment:
didn't we discuss to use a separate service for this, to prevent the framework from having public methods that are not API? A separate service could have `@internal`.

Reply:
well I kinda disagree. I use the `ContaoFramework` class in 1000 places, and I wouldn't want to have a `$this->framework->setLocale()` API, because that is internal. There is no benefit of having this _inside_ the class (apart from having one less class). But it could just as well be in our `LocaleSubscriber`?

Reply:
> But it could just as well be in our `LocaleSubscriber`?

You mean the `LocaleSubscriber` should implement the `LocaleAwareInterface`?

Reply:
I don't know, I would find it odd to add that interface to the event listener. `ContaoFramework` makes still more sense to me.